ansible/roles/test/files/ptftests/arista.py

```diff
@@ -134,10 +134,16 @@ def run(self):
             bgp_neig_output = self.do_cmd('show ip bgp neighbors')
             info['bgp_neig'] = self.parse_bgp_neighbor(bgp_neig_output)
 
-            v4_routing_ok, bgp_route_v4_output = self.check_bgp_route(self.v4_routes)
+            v4_routing, bgp_route_v4_output = self.check_bgp_route(self.v4_routes)
+            if v4_routing != v4_routing_ok:
+                v4_routing_ok = v4_routing
+                self.log('BGP routing for ipv4 OK: %s' % (v4_routing_ok))
```

Comment:
I think no log for bgp routing basically infers that it always returns false. I feel it is okay since no matter how the status changes (even if it is false all the time), we have enough info to trace it via the log message. For the infinite loop problem, I believe it is designed to be like that, and I did not change the logic of this loop in this PR. The usage of the `quit_enabled` flag should be setting the flag and wait for some time until bgp routing to be OK. If it stays in the loop, there should be a timeout on the caller side and report a test failure.